Initial version of file_sorter

Agata-Gizinska opened the pull request:
The aim of this program is to organize files in the current directory according to its extension (file format). 

The program consists of:
- Directories dictionary, which assigns specified file formats into appropriate categories;
- File_Format_Dictionary, which maps file formats into appropriate directories;
- main function, which scans all files in the current directory, checks each file suffix with a dictionary of file formats and creates a new directory for specified file formats.

Agata-Gizinska committed:
Initial version of file sorting program

feat:
- implemented Directories dictionary;
- implemented a dictionary for file formats;
- implemented main function
docs:
- added docstring for the program and for the main function

Agata-Gizinska committed:
Initial version of file sorting program

feat:
- implemented Directories dictionary;
- implemented a dictionary for file formats;
- implemented main function
docs:
- added docstring for the program and for the main function

Agata-Gizinska committed:
Merge remote-tracking branch 'origin' into devwork

czarnadzuma11 commented on clean_my_dir.py:
```diff
@@ -0,0 +1,55 @@
+
+
+if __name__ == "__main__":
+    main()
```
Target directory should be a parameter

Agata-Gizinska replied:
done

Agata-Gizinska committed:
Sort files in prompted directory and use case

feat:
- the program sorts files in prompted directory, not only current directory;
- added test_files for a use case;
docs:
- updated README.md

czarnadzuma11 commented on README.md:
````diff
@@ -25,8 +27,18 @@ Categories:
+argument:
 
-```python clean_my_dir.py```
+```python clean_my_dir.py '\path\to\directory'```
````
"/path/to/directory"
use double quotes so shell can expand strings and use / unix path separators. Your script should be able to handle both

Agata-Gizinska replied:
ok

czarnadzuma11 commented on README.md:
```diff
@@ -25,8 +27,18 @@ Categories:
 ## Usage
 
-Using the terminal change the current directory to the directory with the clean_my_dir.py file. Run command:
+Using the terminal run the program and provide a path to the directory as an
+argument:
```
an argument -> a parameter

Agata-Gizinska replied:
ok

czarnadzuma11 commented on README.md:
````diff
@@ -25,8 +27,18 @@ Categories:
+
+Run the program using the terminal:
+
+```python clean_my_dir.py '.\test_files'```
````
\ -> /

Agata-Gizinska replied:
ok

czarnadzuma11 commented on clean_my_dir.py:
```diff
@@ -0,0 +1,62 @@
+
+
+def main():
+    """The main function of the program. It scans all files in the prompted
+    directory, checks each file suffix with a dictionary of file formats and
```
prompted directory -> specified directory

Agata-Gizinska replied:
ok

czarnadzuma11 commented on clean_my_dir.py:
```diff
@@ -0,0 +1,62 @@
+    directory, checks each file suffix with a dictionary of file formats and
+    creates a new directory for specified file formats."""
+
+    parser = argparse.ArgumentParser(description='Provide directory path')
```
Description string describes the function of the program.

Agata-Gizinska replied:
noted, deleted

czarnadzuma11 commented on clean_my_dir.py:
```diff
@@ -0,0 +1,62 @@
+    parser.add_argument('dir', type=str, help='Directory path')
+    args = parser.parse_args()
+
+    os.chdir(args.dir)
+    for entry in os.scandir():
```
instead of chdir just pass args.dir to os.scandir

Agata-Gizinska replied:
ok

czarnadzuma11 commented on clean_my_dir.py:
```diff
@@ -0,0 +1,62 @@
+        file_path = Path(entry)
+        file_format = file_path.suffix.lower()
+        if file_format in File_Format_Dictionary:
+            directory_path = Path(File_Format_Dictionary[file_format])
```
rename directory_path to dest_directory or such

Agata-Gizinska replied:
ok

czarnadzuma11 commented on clean_my_dir.py:
```diff
@@ -0,0 +1,62 @@
+
+
+def main():
+    """The main function of the program. It scans all files in the prompted
```
Turn this into description string for the program, also mention that it will not sort in subdirectories

Agata-Gizinska replied:
ok

czarnadzuma11 commented on clean_my_dir.py:
```diff
@@ -0,0 +1,62 @@
+    creates a new directory for specified file formats."""
+
+    parser = argparse.ArgumentParser(description='Provide directory path')
+    parser.add_argument('dir', type=str, help='Directory path')
```
Update readme

Agata-Gizinska replied:
ok

Agata-Gizinska committed:
Fixes in the main function

Fix:
- removed chdir
- updated names of variables and paths
Docs:
- updated README.md
- updated doctrings
- added some descriptive comments
